File: src/main/java/gr/apt/controller/AuthenticationController.java
package gr.apt.controller;

import gr.apt.exception.LmsException;
import gr.apt.security.AuthenticationDto;
import gr.apt.security.TokenGenerator;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;

@Path("/auth")
@Produces("application/json")
public class AuthenticationController {

    @Inject
    TokenGenerator tokenGenerator;

    @POST
    @Path("/login")
    @Consumes("application/json")
    public Response login(AuthenticationDto auth) throws LmsException {
        String token = tokenGenerator.generateToken(auth.getUsername());
        if(token != null){
            return Response.ok(token).build();
        }
        throw new LmsException("Could not find user. Please verify your username and password.");
    }
}

File: src/main/java/gr/apt/controller/AuthenticationController.kt
package gr.apt.controller

import gr.apt.config.security.AuthenticationDto
import gr.apt.config.security.TokenGenerator
import javax.inject.Inject
import javax.ws.rs.Consumes
import javax.ws.rs.POST
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.Response

@Path("/auth")
@Produces("application/json")
class AuthenticationController {

    @get:Inject
    lateinit var tokenGenerator: TokenGenerator

    @POST
    @Path("/login")
    @Consumes("application/json")
    fun login(auth: AuthenticationDto): Response {
        val token = tokenGenerator.generateToken(auth.username)
        return Response.ok(token).build()
    }
}